packages/api/src/models/repositories/repository.ts

```diff
@@ -19,6 +19,7 @@ export class Repository extends Model<Repository> implements RepositoryReference
   readable: boolean | undefined;
   writable: boolean | undefined;
   unsupported: boolean | undefined;
+  isNew?: boolean;
```

Comment:
The repository has an `id`. Did you check when the id is assigned and if you can say that the repo is new given the id is still missing?

Reply:
Well. Right. We are inheriting the legacy where the repository name is used as an id and when a repo is created, the name is provided by the user through the UI